Integrations/AWS-EC2/AWS-EC2.py

```diff
@@ -0,0 +1,2679 @@
+    )
+
+    obj = vars(client._client_config)
+
+    response = client.allocate_address(Domain='vpc')
+    data = ({
+        'PublicIp': response['PublicIp'],
+        'AllocationId': response['AllocationId'],
+        'Domain': response['Domain'],
+        'Region': obj['_user_provided_options']['region_name']
+    })
+    ec = {'AWS.EC2.ElasticIPs': data}
+    human_readable = tableToMarkdown('AWS EC2 ElasticIP', data)
+    return_outputs(human_readable, ec)
+
+
+def associate_address_command(args):
+    client = aws_session(
+        service='ec2',
+        region=args.get('region'),
+        roleArn=args.get('roleArn'),
+        roleSessionName=args.get('roleSessionName'),
+        roleSessionDuration=args.get('roleSessionDuration'),
+    )
+
+    obj = vars(client._client_config)
+    kwargs = {'AllocationId': args.get('allocationId')}
+
+    if args.get('instanceId') is not None:
+        kwargs.update({'InstanceId': args.get('instanceId')})
+    if args.get('allowReassociation') is not None:
+        kwargs.update({'AllowReassociation': True if args.get('allowReassociation') == 'True' else False})
+    if args.get('networkInterfaceId') is not None:
+        kwargs.update({'NetworkInterfaceId': args.get('networkInterfaceId')})
+    if args.get('privateIpAddress') is not None:
+        kwargs.update({'PrivateIpAddress': args.get('privateIpAddress')})
+
+    response = client.associate_address(**kwargs)
+    data = ({
+        'AllocationId': args.get('allocationId'),
+        'AssociationId': response['AssociationId'],
+        'Region': obj['_user_provided_options']['region_name']
+    })
+
+    ec = {"AWS.EC2.ElasticIPs(val.AllocationId === obj.AllocationId)": data}
+    human_readable = tableToMarkdown('AWS EC2 ElasticIP', data)
+    return_outputs(human_readable, ec)
+
+
+def create_snapshot_command(args):
+    client = aws_session(
+        region=args.get('region'),
+        roleArn=args.get('roleArn'),
+        roleSessionName=args.get('roleSessionName'),
+        roleSessionDuration=args.get('roleSessionDuration'),
+    )
+    obj = vars(client._client_config)
+    kwargs = {'VolumeId': args.get('volumeId')}
+
+    if args.get('description') is not None:
+        kwargs.update({'Description': args.get('description')})
+    if args.get('tags') is not None:
+        kwargs.update({
+            'TagSpecifications': [{
+                'ResourceType': 'snapshot',
+                'Tags': parse_tag_field(args.get('tags'))}]
+        })
+
+    response = client.create_snapshot(**kwargs)
+
+    data = ({
+        'Description': response['Description'],
+        'Encrypted': response['Encrypted'],
+        'Progress': response['Progress'],
+        'SnapshotId': response['SnapshotId'],
+        'State': response['State'],
+        'VolumeId': response['VolumeId'],
+        'VolumeSize': response['VolumeSize'],
+        'StartTime': datetime.strftime(response['StartTime'], '%Y-%m-%dT%H:%M:%SZ'),
```

Comment:
this can throw value error. better calculate it before the append and in a try block.